napari/_qt/qt_resources/_icons.py

```diff
@@ -0,0 +1,185 @@
+import os
+from contextlib import contextmanager
+from itertools import product
+from pathlib import Path
+from tempfile import NamedTemporaryFile, TemporaryDirectory
+from typing import Generator, Iterable, Optional
+
+import qtpy
+
+QRC_TEMPLATE = """
+<!DOCTYPE RCC><RCC version="1.0">
+<qresource prefix="/themes">
+{}
+</qresource>
+</RCC>
+"""
+
+
+DEFAULT_OPACITIES = (0.5, 1)
+
+
+@contextmanager
+def temporary_qrc_file(
+    themes: Optional[Iterable[str]] = None,
+    icons: Optional[Iterable[str]] = None,
+    opacities: Optional[Iterable[float]] = None,
+) -> Generator[str, None, None]:
+    """Create a qrc file for all combinations of ``icon``, ``theme``, ``opac``.
+
+    Generates a .qrc file containing, for all theme/icons:
+
+    <qresource prefix="/themes">
+        <file alias='theme/icon_name.svg'>theme_icon_name.svg</file>
+    </qresource>
+
+    Parameters
+    ----------
+    themes : iterable of str, optional
+        The theme names for which to generate icons, by default all available
+        themes.
+    icons : iterable of str, optional
+        The set of icons , by default all available icons
+    opacities : iterable of float, optional
+        An iterable of icon opacities to pre-generate, by default (0.6, 1).
+        Icons with opacity < 1 can be accessed with the opacity as a percentage
+        for instance: ``icon_name_50.svg``
+
+    Yields
+    -------
+    QRC_name : str
+        The name of the temporary QRC file.  It will be in a temporary
+        directory containing all of the necessary modified/colorized svgs.
+    """
+    from ...resources._icons import ICONS, get_colorized_svg
+    from ...utils.theme import _themes
+
+    themes = themes or _themes.keys()
+    icons = icons or ICONS.keys()
+    opacities = opacities or DEFAULT_OPACITIES
+
+    # theme, icon_name, opacity_key, colorized.svg
+    FILE_T = "<file alias='{}/{}{}.svg'>{}</file>"
+
+    # mapping of icon_name to theme key, otherwise icon color will be "icon"
+    color_override = {'warning': 'warning'}
+
+    # create a temporary directory for the qrc file and all colorized svgs.
+    with TemporaryDirectory() as tdir_name:
+        tmp_dir = Path(tdir_name)
+
+        # create the colorized SVGs
+        files = []
+        for theme, icon_name, op in product(_themes, ICONS, opacities):
```

Comment:
```suggestion
        for theme, icon_name, op in product(themes, icons, opacities):
```

Maybe ? or your variables lines 57 are unused ?

Reply:
sorry, I'm about to submit a refactored version of this... will clean this up (the gist is the same though)